watchdog/src/bitcoin_block_apis.rs

```diff
@@ -1,25 +1,45 @@
 use crate::endpoints::*;
+use ic_cdk::api::management_canister::http_request::HttpResponse;
 use serde_json::json;
 
 /// APIs that serve Bitcoin block data.
-#[derive(Debug)]
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
 pub enum BitcoinBlockApi {
-    // TODO: investigate why this API is not working.
-    #[allow(dead_code)]
     ApiBitapsCom,
-
     ApiBlockchairCom,
     ApiBlockcypherCom,
-    BitcoinCanister,
+    BitcoinCanister, // Not an explorer.
```

Comment:
Yeah this part is a bit weird. I would actually expect that we can make a simple inter-canister call to get the height from the bitcoin canister (of course we need to add a method that returns it first). I don't think we should take action immediately but the fact we have to add a comment to clarify is a bit of a code smell.